### PR TITLE
bugfix(catalog-backend-module-gitlab): Make sure archived repos are skipped

### PR DESCRIPTION
**Pull request by ivangonzalezacuna**
## Hey, I just made a Pull Request!

This error appeared since the archived repos are filtered via the GitLab API in the PR https://github.com/backstage/backstage/pull/18816. However, this never worked properly.

For some time we noticed a bunch of archived repos were fetched again by Backstage, and after investigating the reason, we discovered what is the reason:

In the `pagedRequest` function _only_ the values that passed the `options[key]` were accepted.  The `archived` parameter was always set to `false` so this check never passed and the generated URL would look like this: `<GITLAB_URL>/groups/<GROUP>/projects?group=<GROUP>&page=1&per_page=50&include_subgroups=true`. When this is called, the GitLab API is also returning the archived repos. 

Therefore, we have to make sure that _only_ the empty options are skipped (the ones that are an empty string). Once this is done, the URL will be like the following: `<GITLAB_URL>/groups/<GROUP>/projects?archived=false&group=<GROUP>&page=1&per_page=50&include_subgroups=true`. Which is now skipping the archived repos.

I also needed to update a test, because an archived repo was returned and it was causing the test to fail. Removing that archived repo fixed it.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/catalog-backend-module-gitlab/src/lib/client.ts [hunk 1]
```diff
@@ -309,7 +309,7 @@ export class GitLabClient {
   ): Promise<PagedResponse<T>> {
     const request = new URL(`${this.config.apiBaseUrl}${endpoint}`);
     for (const key in options) {
-      if (options[key]) {
+      if (options[key]?.toString()) {
```
Comment (jhaals) [inline plugins/catalog-backend-module-gitlab/src/lib/client.ts:312]:
```suggestion
      if (options[key] !== undefined) {
```

I believe it's a bit clearer to check for not undefined instead.

**Review by jhaals — CHANGES_REQUESTED** [review]
Thanks for fixing this! It would be great to actually have a test that verifies that we pass the options properly.

File: plugins/catalog-backend-module-gitlab/src/lib/client.ts (hunk 1 above)
Comment (ivangonzalezacuna) [reply plugins/catalog-backend-module-gitlab/src/lib/client.ts:312]: Will test it tomorrow morning, just to be sure it works still. But yeah, that makes it a bit clearer

**Comment by ivangonzalezacuna** [comment]
> Thanks for fixing this! It would be great to actually have a test that verifies that we pass the options properly.

That would be a great idea! The best solution I can think of at the moment might be creating a fake server that simply returns the URL that was used for the call, and then compare it with the expected one. Basically, this was working for all the scenarios except for the values in the `CommonListOptions` which were booleans. Will implement that test tomorrow morning

File: plugins/catalog-backend-module-gitlab/src/lib/client.ts (hunk 1 above)
Comment (ivangonzalezacuna) [reply plugins/catalog-backend-module-gitlab/src/lib/client.ts]: I came up with a different solution. If checking for undefined, an empty string would be allowed. So I'm checking it's not undefined and neither an empty string. I added some tests as well that check that behavior

**Review by Rugvip — APPROVED** [review]
Good find! 👍 

Thank you for adding tests as well 🎉

**Pull request merged**